pkg/pipeline/extract/aggregate/aggregates.go

```diff
@@ -54,20 +63,39 @@ func (aggregates Aggregates) GetMetrics() []config.GenericMap {
 func (aggregates Aggregates) AddAggregate(aggregateDefinition api.AggregateDefinition) Aggregates {
 	aggregate := Aggregate{
 		Definition: aggregateDefinition,
-		Groups:     map[NormalizedValues]*GroupState{},
+		GroupsMap:  map[NormalizedValues]*GroupState{},
+		GroupsList: list.New(),
 	}
 
 	appendedAggregates := append(aggregates, aggregate)
 	return appendedAggregates
 }
 
-func (aggregates Aggregates) RemoveAggregate(by api.AggregateBy) (Aggregates, error) {
```

Comment:
When was `RemoveAggregate` ever called? No longer relevant.

Reply:
right, it is not needed this is why I removed